swupd/create_manifests_test.go

```diff
@@ -398,3 +398,38 @@ func TestCreateManifestIncludeVersionBump(t *testing.T) {
 	mustNotExist(t, filepath.Join(testDir, "www/30/Manifest.test-bundle"))
 	mustNotExist(t, filepath.Join(testDir, "www/30/Manifest.included2"))
 }
+
+func TestCreateManifestsState(t *testing.T) {
+	testDir := mustSetupTestDir(t, "state")
+	defer removeIfNoErrors(t, testDir)
+	mustInitStandardTest(t, testDir, "0", "10", []string{})
+	mustGenFile(t, testDir, "10", "os-core", "var/lib/test", "test")
+	if err := CreateManifests(10, false, 1, testDir); err != nil {
+		t.Fatal(err)
+	}
+
+	re := regexp.MustCompile("D\\.s\\.\t.*\t10\t/var/lib\n")
```

Comment:
Here we are doing a very textual test, seeing that we have a particular string (well regular expression) in the file. Would it be better to have a more logical test, saying do we have a File{Flags: something, Version: 10, Name: "/var/lib"} in the manifest? This would make the test independent of the file format.

Reply:
Well, I do want to check the *format* of the text manifest, since that is important to us at this point. I do like the idea of abstracting that away though, so a test can just call the function to test that a file exists in the manifest and the function can do the heavy lifting of checking the format in the manifest file.

Reply:
I'm going to defer this one to later, since I have some other pressing things to work on.